lib/importer.js

```diff
@@ -3,17 +3,51 @@ var _ = require('underscore'),
     rsvp = require('rsvp'),
     format = require('pg-format');
 
+var DEFAULT_DATABASE = 'default-source';
+
+var DELETE_STMT = 'DELETE FROM couchdb WHERE doc->>\'_id\' in (%L)';
+var SELECT_SEQ_STMT = 'SELECT seq FROM couchdb_progress WHERE source = %L';
+var UPDATE_SEQ_SOURCE_STMT = 'UPDATE couchdb_progress SET source = %L WHERE source = %L';
+var INSERT_SEQ_STMT = 'INSERT INTO couchdb_progress(seq, source) VALUES (%L, %L)';
+var UPDATE_SEQ_STMT = 'UPDATE couchdb_progress SET seq = %L WHERE source = %L';
+var INSERT_DOC_STMT = 'INSERT INTO couchdb (doc) VALUES %L';
+
 var deleteDocuments = function(db, docIdsToDelete) {
   if (docIdsToDelete && docIdsToDelete.length) {
-    return db.query(
-      format('DELETE FROM couchdb WHERE doc->>\'_id\' in (%L)', docIdsToDelete));
+    return db.query(format(DELETE_STMT, docIdsToDelete));
   } else {
     return rsvp.Promise.resolve();
   }
 };
 
-var storeSeq = function(db, seq) {
-  return db.query(format('UPDATE couchdb_progress SET seq = %L', seq));
+var getSeq = function(db, source) {
+  // Find sequence for source
+  return db.one(format(SELECT_SEQ_STMT, source))
+    .then(function(result) {
+      return result.seq;
+    })
+    .catch(function() {
```

Comment:
You probably want to check these errors. IDK what Postgres will send back, but ideally you can tell the difference between "your SQL invalid", "the DB isn't running" and "could not get one result", and proceed accordingly.